src/denyList.ts

```diff
@@ -0,0 +1,102 @@
+import { Redis } from "./types"
+import { RatelimitResponse } from "./types"
+import { Cache } from "./cache";
+
+
+const denyListCache = new Cache(new Map());
+
+/**
+ * Checks items in members list and returns true if any
+ * of them are in denyListCache.
+ * 
+ * @param members list of values to check against the cache
+ * @returns a member from the cache. If there is none, returns undefined
+ */
+export const checkDenyListCache = (members: string[]): string | undefined => {
+  return members.filter(
+    member => denyListCache.isBlocked(member).blocked
+  )[0]
+}
+
+/**
+ * Blocks a member for 1 minute.
+ * 
+ * If there are more than 1000 elements in the cache, empties
+ * it so that the cache doesn't grow in size indefinetely.
+ * 
+ * @param member member to block
+ */
+const blockMember = (member: string) => {
+  if (denyListCache.size() > 1000) denyListCache.empty();
+  denyListCache.blockUntil(member, Date.now() + 60000);
+}
+
+/**
+ * Checks if identifier or any of the values are in any of
+ * the denied lists in Redis.
+ * 
+ * If some value is in a deny list, we block the identifier for a minute.
+ * 
+ * @param redis redis client
+ * @param prefix ratelimit prefix
+ * @param members List of values (identifier, ip, user agent, country)
+ * @returns 
+ */
+export const checkDenyList = async (
+  redis: Redis,
+  prefix: string,
+  members: string[]
+) => {
+  const deniedMembers = await redis.smismember(
+    [prefix, "denyList", "all"].join(":"),
+    members
+  );
+
+  let deniedMember: string | undefined = undefined;
+  deniedMembers.map((memberDenied, index) => {
```

Comment:
Maybe we can try a simple `for-of` instead of map, since we are not actually mapping anything?